matdeeplearn/common/data.py

```diff
@@ -71,39 +74,35 @@ def get_dataset(
     data_path: str
         path to the folder containing data.pt file
 
-    target_index: int
-        the index to select the target values
-        this is needed because in our target.csv, there might be
-        multiple columns of target values available for that
-        particular dataset, thus we need to index one column for
-        the current run/experiment
-
-    transform_type: transformation function/class to be applied
+    transform_list: transformation function/classes to be applied
     """
 
+    transforms = []
     # set transform method
-    if transform_type == "GetY":
-        T = GetY
-    else:
-        raise ValueError("No such transform found for {transform}")
+    for transform in transform_list:
+        if transform["otf"]:
```

Comment:
Let's make this optional, so default behavior is False
```suggestion
        if transform.get("otf", False):
```